src/main/java/io/stargate/sgv2/jsonapi/service/shredding/tables/WriteableTableRow.java

```diff
@@ -1,15 +1,111 @@
 package io.stargate.sgv2.jsonapi.service.shredding.tables;
 
-import com.datastax.oss.driver.api.core.CqlIdentifier;
-import io.stargate.sgv2.jsonapi.service.shredding.DocRowIdentifer;
-import io.stargate.sgv2.jsonapi.service.shredding.WritableDocRow;
-import java.util.Map;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.TableSchemaObject;
+import io.stargate.sgv2.jsonapi.service.shredding.*;
+import io.stargate.sgv2.jsonapi.util.PrettyPrintable;
+import io.stargate.sgv2.jsonapi.util.PrettyToStringBuilder;
+import java.util.Objects;
 
-public record WriteableTableRow(RowId id, Map<CqlIdentifier, Object> allColumnValues)
-    implements WritableDocRow {
+/**
+ * Data that is ready to be written to a CQL table, it has been validated and we expect the insert
+ * to work.
+ *
+ * <p>NOTE: create these using the {@link
+ * io.stargate.sgv2.jsonapi.service.operation.tables.WriteableTableRowBuilder} it knows how to
+ * validate and convert the data.
+ *
+ * <p><b>INVARIANTS:</b> For this class to be valid, the following must be true:
+ *
+ * <ul>
+ *   <li>All the key and non primary key columns exist in the target table
+ *   <li>All key columns have values.
+ *   <li>All types have been converted to what he driver expects.
+ * </ul>
+ */
+public class WriteableTableRow implements WritableDocRow, PrettyPrintable {
```

Comment:
We have gone back and forth, I think, but `WritableDocRow` would imply Document (and hence Collection). Maybe it should be just `WritableRow`, with sub-class of `WritableTableRow` instead?